### PR TITLE
feat(nextjs): Accept `redirectUrl` as option for `auth().protect()`

### PR DESCRIPTION
**Pull request by panteliselef**
## Description

This PR allows developers to call `auth().protect()` and pass to it a `redirectUrl` which can be used as the fallback instead of the standard 404 error that we currently throw.

```ts
import { auth } from '@clerk/nextjs/server';

// Authorization
auth().protect({ role:'org:admin' }, { redirectUrl:'/any-page' })
auth().protect({ permission:'org:settings:manage' }, { redirectUrl:'/any-page' })

// Authentication
auth().protect({ redirectUrl:'/any-page' })
```

## Checklist

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [x] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [ ] `@clerk/backend`
- [ ] `@clerk/chrome-extension`
- [ ] `@clerk/clerk-js`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/fastify`
- [ ] `gatsby-plugin-clerk`
- [ ] `@clerk/localizations`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/clerk-react`
- [ ] `@clerk/remix`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/shared`
- [ ] `@clerk/themes`
- [ ] `@clerk/types`
- [ ] `build/tooling/chore`

### CODE REVIEW

**Comment by panteliselef** [comment]
@BRKalow @nikosdouvlis Do you find this too annoying ? Types for params are already too complex.
```ts
auth().protect(undefined, { redirectUrl:'/any-page' })
```

**Review by chanioxaris — APPROVED** [review]

**Comment by panteliselef** [comment]
Avoid merging before the question above is answered

**Comment by BRKalow** [comment]
@panteliselef Is the alternative to adjust the types such that this is possible?

```ts
auth().protect({ redirectUrl: '/' })
```

**Review by dimkl — APPROVED** [review]

File: .changeset/thirty-chicken-divide.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@clerk/nextjs': patch
+---
+
+Accept `redirectUrl` as option for `auth().protect()`
```
Comment (LekoArts) [inline .changeset/thirty-chicken-divide.md:5]:
````suggestion
Accept `redirectUrl` as option for `auth().protect()`.

For example:

```ts
// Authorization
auth().protect({ role:'org:admin' }, { redirectUrl:'/any-page' })
auth().protect({ permission:'org:settings:manage' }, { redirectUrl:'/any-page' })

// Authentication
auth().protect({ redirectUrl:'/any-page' })
```
````

**Review by BRKalow — APPROVED** [review]
👏

**Pull request merged**